code/datums/components/obj_projectile_damage.dm

```diff
@@ -0,0 +1,31 @@
+/datum/component/obj_projectile_damage
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+/datum/component/obj_projectile_damage/Initialize()
+	if(!istype(parent, /obj))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, COMSIG_ATOM_HITBY_PROJ, .proc/projectile_collide)
+	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, .proc/examine)
+
+/datum/component/obj_projectile_damage/proc/projectile_collide(owner, var/obj/projectile/P)
+	var/obj/O = parent
+	if(P.proj_data.damage_type & (D_KINETIC | D_ENERGY | D_SLASHING))
+		if (O && (O._health/O._max_health) <= 0.5 && prob((O._health/O._max_health) * 60))
```

Comment:
prob should go up as health-fraction goes down, shouldn't it??

Reply:
I assume Mordent wanted that changed